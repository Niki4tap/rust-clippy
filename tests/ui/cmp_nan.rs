#![warn(clippy::cmp_nan)]
#![allow(clippy::float_cmp, clippy::no_effect, clippy::unnecessary_operation)]

const NAN_F32: f32 = f32::NAN;
const NAN_F64: f64 = f64::NAN;

// don't lint these, since `{f32,f64}::is_nan` is not yet const stable
// FIXME: make this lint when `{f32,f64}::is_nan` becomes const stable
const IS_F32_NAN: bool = NAN_F32 == f32::NAN;
const IS_F64_NAN: bool = NAN_F64 == f64::NAN;

fn main() {
    let x = 5f32;
    x == f32::NAN;
    x != f32::NAN;
    x < f32::NAN;
    x > f32::NAN;
    x <= f32::NAN;
    x >= f32::NAN;
    x == NAN_F32;
    x != NAN_F32;
    x < NAN_F32;
    x > NAN_F32;
    x <= NAN_F32;
    x >= NAN_F32;

    let y = 0f64;
    y == f64::NAN;
    y != f64::NAN;
    y < f64::NAN;
    y > f64::NAN;
    y <= f64::NAN;
    y >= f64::NAN;
    y == NAN_F64;
    y != NAN_F64;
    y < NAN_F64;
    y > NAN_F64;
    y <= NAN_F64;
    y >= NAN_F64;
}
