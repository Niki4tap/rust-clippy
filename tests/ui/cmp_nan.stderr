error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:14:5
   |
LL |     x == f32::NAN;
   |     ^^^^^^^^^^^^^
   |
   = note: `-D clippy::cmp-nan` implied by `-D warnings`

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:15:5
   |
LL |     x != f32::NAN;
   |     ^^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:16:5
   |
LL |     x < f32::NAN;
   |     ^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:17:5
   |
LL |     x > f32::NAN;
   |     ^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:18:5
   |
LL |     x <= f32::NAN;
   |     ^^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:19:5
   |
LL |     x >= f32::NAN;
   |     ^^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:20:5
   |
LL |     x == NAN_F32;
   |     ^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:21:5
   |
LL |     x != NAN_F32;
   |     ^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:22:5
   |
LL |     x < NAN_F32;
   |     ^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:23:5
   |
LL |     x > NAN_F32;
   |     ^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:24:5
   |
LL |     x <= NAN_F32;
   |     ^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:25:5
   |
LL |     x >= NAN_F32;
   |     ^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:28:5
   |
LL |     y == f64::NAN;
   |     ^^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:29:5
   |
LL |     y != f64::NAN;
   |     ^^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:30:5
   |
LL |     y < f64::NAN;
   |     ^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:31:5
   |
LL |     y > f64::NAN;
   |     ^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:32:5
   |
LL |     y <= f64::NAN;
   |     ^^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:33:5
   |
LL |     y >= f64::NAN;
   |     ^^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:34:5
   |
LL |     y == NAN_F64;
   |     ^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:35:5
   |
LL |     y != NAN_F64;
   |     ^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:36:5
   |
LL |     y < NAN_F64;
   |     ^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:37:5
   |
LL |     y > NAN_F64;
   |     ^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:38:5
   |
LL |     y <= NAN_F64;
   |     ^^^^^^^^^^^^

error: doomed comparison with `NAN`, use `{f32,f64}::is_nan()` instead
  --> $DIR/cmp_nan.rs:39:5
   |
LL |     y >= NAN_F64;
   |     ^^^^^^^^^^^^

error: aborting due to 24 previous errors

